debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java

```diff
@@ -127,8 +127,14 @@ static String currentScnQuery() {
         return String.format("SELECT CURRENT_SCN FROM %s", DATABASE_VIEW);
     }
 
-    static String oldestFirstChangeQuery() {
-        return String.format("SELECT MIN(FIRST_CHANGE#) FROM %s", LOG_VIEW);
+    static String oldestFirstChangeQuery(long archiveLogDays) {
+        if (archiveLogDays > 0) {
+            return String.format("SELECT MIN(FIRST_CHANGE#) FROM (SELECT MIN(FIRST_CHANGE#) AS FIRST_CHANGE# FROM %s " +
+                    "UNION SELECT MIN(FIRST_CHANGE#) AS FIRST_CHANGE# FROM %s " +
+                    "WHERE FIRST_TIME >= SYSDATE - %d)", LOG_VIEW, ARCHIVED_LOG_VIEW, archiveLogDays);
```

Comment:
does SYSDATE - <days> actually work? Does this not need some explicit conversion into a date type?

Reply:
No conversion is needed.  Oracle interprets `SYSDATE` as the current date automatically and any addition or subtraction to that value is interpreted as +/- number of whole days.  If we wanted the last X hours, you'd write it as `SYSDATE - (%d/24)` or use the interval syntax of `SYSDATE - INTERVAL '%d' hours`.

Reply:
Ok, I see. Thx!